src/main/java/io/vertx/ext/stomp/impl/Queue.java

```diff
@@ -0,0 +1,216 @@
+package io.vertx.ext.stomp.impl;
+
+import io.vertx.core.Vertx;
+import io.vertx.ext.stomp.Destination;
+import io.vertx.ext.stomp.Frame;
+import io.vertx.ext.stomp.StompServerConnection;
+import io.vertx.ext.stomp.utils.Headers;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+/**
+ * Implementation of {@link Destination} dispatching messages to a single subscriber. It dispatches
+ * the messages using a round-robin strategy.
+ *
+ * @author <a href="http://escoffier.me">Clement Escoffier</a>
+ */
+public class Queue implements Destination {
+
+  private final String destination;
+
+  private final List<Subscription> subscriptions = new ArrayList<>();
+  private final Vertx vertx;
+  private int lastUsedSubscriptions = -1;
+
+  public Queue(Vertx vertx, String destination) {
+    this.destination = destination;
+    this.vertx = vertx;
+  }
+
+  /**
+   * @return the destination address.
+   */
+  @Override
+  public String destination() {
+    return destination;
+  }
+
+  /**
+   * Dispatches the given frame.
+   *
+   * @param connection the connection
+   * @param frame      the frame ({@code SEND} frame).
+   * @return the current instance of {@link Destination}
+   */
+  @Override
+  public synchronized Destination dispatch(StompServerConnection connection, Frame frame) {
+    if (subscriptions.isEmpty()) {
+      lastUsedSubscriptions = -1;
+      return this;
+    }
+    Subscription subscription = getNextSubscription();
```

Comment:
What happens if a message is dispatched to a destination but there are no subscriptions for it?

In this case, in most messaging implementations the queue would queue the message and deliver it to a subscription that may subscribe later.